executor/insert_common.go

```diff
@@ -547,15 +565,29 @@ func (e *InsertValues) doBatchInsert(ctx context.Context) error {
 func (e *InsertValues) getRow(ctx context.Context, vals []types.Datum) ([]types.Datum, error) {
 	row := make([]types.Datum, len(e.Table.Cols()))
 	hasValue := make([]bool, len(e.Table.Cols()))
+	sc := e.ctx.GetSessionVars().StmtCtx
+	warnCnt := int(sc.WarningCount())
+
 	for i := 0; i < e.rowLen; i++ {
-		casted, err := table.CastValue(e.ctx, vals[i], e.insertColumns[i].ToInfo(), false, false)
-		if e.handleErr(nil, &vals[i], 0, err) != nil {
+		col := e.insertColumns[i].ToInfo()
+		casted, err := table.CastValue(e.ctx, vals[i], col, false, false)
+		if err = e.handleErr(e.insertColumns[i], &vals[i], int(e.rowCount), err); err != nil {
 			return nil, err
 		}
 
 		offset := e.insertColumns[i].Offset
 		row[offset] = casted
 		hasValue[offset] = true
+
+		if e.ctx.GetSessionVars().StmtCtx.InLoadDataStmt {
+			if newWarnings := sc.TruncateWarnings(warnCnt); len(newWarnings) > 0 {
```

Comment:
on which case we need to handle existing warnings?

Reply:
to fix 43555, we need to rewrite the warnings

Reply:
I copied from https://github.com/pingcap/tidb/blob/cd834eec65c54fdae2869adc2d76f3f8f7868042/executor/insert.go#L415-L422